Assignments/SCF-Session-2/hexadecimalCalc/HexCalc.java

```diff
@@ -0,0 +1,277 @@
+	{
+		String digits = "0123456789ABCDEF";
+		hexNum = hexNum.toUpperCase();  
+        	int value = 0;  
+        	for (int i = 0; i < hexNum.length(); i++)  
+        	{
+			char character = hexNum.charAt(i);  
+            		int digit = digits.indexOf(character);  
+            		value = 16 * value + digit;  
+        	}  
+        	return value;  
+	}
+	
+	public String decToHex(int decNum)
+	{
+		int remender;  
+	    	String hex = "";   
+	    	char hexchars[] = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F'};  
+	    	while(decNum > 0)  
+	    	{
+	    		remender = decNum % 16;   
+	        	hex = hexchars[remender] + hex;   
+	        	decNum = decNum / 16;  
+	    	}  
+	    	return hex;  
+	}
+	
+	public boolean greater(String hexNum1, String hexNum2)
+	{
+		hexNum1 = removeZeros(hexNum1);
+		hexNum2 = removeZeros(hexNum2);
+		boolean greaterNum  = false;
+		String digits = "0123456789ABCDEF";
+		hexNum1 = hexNum1.toUpperCase();
+		hexNum2 = hexNum2.toUpperCase();
+		
+		if(hexNum1.length() > hexNum2.length())
+			greaterNum = true;
+		else 
+			if(hexNum1.length() == hexNum2.length())
+		    	{
+				for(int i = 0; i < hexNum1.length(); i++)
+				{
+					if(digits.indexOf(hexNum1.charAt(i)) > digits.indexOf(hexNum2.charAt(i)))
+					{
+						greaterNum = true;
+						break;
+					}
+					else if(hexNum1.charAt(i) == hexNum2.charAt(i))
+						continue;
+					else
+						break;
+				}
+		    	}
+			return greaterNum;
+	}
+	
+	public boolean smaller(String hexNum1, String hexNum2)
+	{
+		hexNum1 = removeZeros(hexNum1);
+		hexNum2 = removeZeros(hexNum2);
+		boolean smallerNum  = false;
+		String digits = "0123456789ABCDEF";
+		hexNum1 = hexNum1.toUpperCase();
+		hexNum2 = hexNum2.toUpperCase();
+		
+		if(hexNum1.length() < hexNum2.length())
+			smallerNum = true;
+		else 
+			if(hexNum1.length() == hexNum2.length())
+		    	{
+				for(int i = 0; i < hexNum1.length(); i++)
+				{
+					if(digits.indexOf(hexNum1.charAt(i)) < digits.indexOf(hexNum2.charAt(i)))
+					{
+						smallerNum = true;
+						break;
+					}
+					else if(hexNum1.charAt(i) == hexNum2.charAt(i))
+						continue;
+					else
+						break;
+				}
+		    	}
+			return smallerNum;
+	}
+	
+	public boolean equality(String hexNum1, String hexNum2)
+	{
+		hexNum1 = removeZeros(hexNum1);
+		hexNum2 = removeZeros(hexNum2);
+		boolean equal = false;
+		
+		if(hexNum1.equals(hexNum2))
+			equal = true;
+		return equal;
+	}
+	
+	public String removeZeros(String hexNum)
+	{
+		int i = 0;
+		String newString = "";
+		while(hexNum.charAt(i) == '0')
+			i++;
+		while(i	< hexNum.length())
+		{
+			newString += hexNum.charAt(i);
+			i++;
+		}
+		
+		return newString;
+	}
+}
+
+
+public class HexCalc 
+{
+	public static void main(String[] args)
+	{
+		int decNum, choice;
+		String hexNum, hexNum1, hexNum2;
+		
+		Functions fun = new Functions();
+		
+		Scanner sc = new Scanner(System.in);
+		
+		do
+		{
+			System.out.println("MENU");
```

Comment:
New line character `\n` should be used instead of multiple `System.out.println()`.

Reply:
done